### PR TITLE
 update for link tokens 

### PR DESCRIPTION
**Pull request by davidzhanghp**

### CODE REVIEW

File: client/src/services/link.js [hunk 1]
```diff
@@ -44,104 +43,89 @@ export function LinkProvider(props) {
...
+          } else {
+            await exchangeToken(publicToken, institution, accounts, userId);
+            getItemsByUser(userId, true);
+          }
+
+          if (window.location.pathname === '/') {
+            window.location.href = `/user/${userId}/items`;
+          }
+        },
+        onExit: async (
```
Comment (otherchen) [inline client/src/services/link.js:99]: In this `onExit` handler, perhaps we should show the best practice of handling the `INVALID_LINK_TOKEN` error_code when a link_token is invalidated (too many attempts, expired, etc).

```
if (error != null && error.error_code === 'INVALID_LINK_TOKEN`) {
    // gets rid of the old iframe
    handler.destroy();
    // generate a new token and reinitialize Link
    handler.create({... oldConfigs, token: await getLinkToken({ userId, itemId }) });
}
```

File: client/src/services/link.js [hunk 2]
```diff
@@ -151,10 +135,9 @@ export function LinkProvider(props) {
    */
   const getLinkHandler = useCallback(
     ({ userId, itemId } = {}) => {
-      if (itemId && !hasRequested.current.byItem[itemId]) {
-        createLinkHandler({ itemId });
-      } else if (userId && !hasRequested.current.byUser[userId]) {
-        createLinkHandler({ userId });
+      console.log('user1', userId, 'item1', itemId);
```
Comment (otherchen) [inline client/src/services/link.js:138]: nit - extra log

File: client/src/services/webhooks.js [hunk 3]
```diff
@@ -1,49 +0,0 @@
-import React, {
```
Comment (otherchen) [inline client/src/services/webhooks.js:1]: I wonder why we don't need this anymore 🤔

File: server/routes/linkTokens.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+const {
+  retrieveItemById,
+} = require('../db/queries');
+
+const router = express.Router();
+
+router.post('/', asyncWrapper(async (req, res) => {
+  try {
+    const { userId, itemId } = req.body;
+    console.log('hm', JSON.stringify(req.body), itemId);
```
Comment (otherchen) [inline server/routes/linkTokens.js:19]: extra log

File: server/routes/linkTokens.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    if (itemId != null) {
+      const itemIdResponse = await retrieveItemById(itemId);
+      console.log('response', JSON.stringify(itemIdResponse));
+      accessToken = itemIdResponse.plaid_access_token;
+      products = [];
+    }
+    const response = await fetch('http://ngrok:4040/api/tunnels');
+    const { tunnels } = await response.json();
+    const httpTunnel = tunnels.find(t => t.proto === 'http');
+    console.log('tunenls', JSON.stringify(httpTunnel));
```
Comment (otherchen) [inline server/routes/linkTokens.js:31]: `tunnels`

File: client/src/services/webhooks.js (hunk 3 above)
Comment (davidzhanghp) [reply client/src/services/webhooks.js:1]: we used to hit an endpoint to get the webhook to pass in link. since we do it server side we no longer need it

**Comment by davidzhanghp** [comment]
also addresses #22

**Pull request merged**